src/hooks/custom-hooks/useOAuth.ts

```diff
@@ -37,18 +38,28 @@ const useOAuth = (): UseOAuthReturn => {
     const authTokenLocalStorage = localStorage.getItem('authToken');
 
     const WSLogoutAndRedirect = async () => {
-        await logout();
-        removeCookies('affiliate_token', 'affiliate_tracking', 'utm_data', 'onfido_token', 'gclid');
-        if (!isOAuth2Enabled) {
-            window.open(oauthUrl, '_self');
+        try {
+            await logout();
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error('Failed to logout', error);
         }
+        removeCookies('affiliate_token', 'affiliate_tracking', 'utm_data', 'onfido_token', 'gclid');
+        redirectToAuth();
     };
+
     const handleLogout = async () => {
-        await OAuth2Logout({
-            postLogoutRedirectUri: window.location.origin,
-            redirectCallbackUri: `${window.location.origin}/callback`,
-            WSLogoutAndRedirect,
-        });
+        try {
+            await OAuth2Logout({
+                postLogoutRedirectUri: window.location.origin,
+                redirectCallbackUri: `${window.location.origin}/callback`,
+                WSLogoutAndRedirect,
+            });
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error('Failed to handle logout', error);
+            showErrorModal?.();
```

Comment:
which error are we showing here?

Reply:
This will show the OIDC-related errors, which can be picked up by Datadog
![Screenshot 2025-03-26 at 2 56 42 PM](https://github.com/user-attachments/assets/b4ee8d19-e813-4101-bf66-9dc92b68ab42)

Reply:
we're showing this error

![Screenshot 2025-03-26 at 2 01 28 PM](https://github.com/user-attachments/assets/9e5d56d7-1861-465c-b835-4821278604f9)